packages/gatsby-source-npm/gatsby-node.js

```diff
@@ -0,0 +1,74 @@
+const algoliasearch = require(`algoliasearch`)
+const crypto = require(`crypto`)
+
+var client = algoliasearch(`OFCNCOG2CU`, `f54e21fa3a2a0160595bb058179bfb1e`)
+var index = client.initIndex(`npm-search`)
+
+const createContentDigest = obj =>
+  crypto
+    .createHash(`md5`)
+    .update(JSON.stringify(obj))
+    .digest(`hex`)
+
+exports.sourceNodes = async ({ boundActionCreators }) => {
+  const { createNode } = boundActionCreators
+
+  console.log(`Grabbing NPM packages...`)
+
+  const data = await index.search({
```

Comment:
This should be index.browseAll() or browse, see how it's done with https://github.com/algolia/algolia-sitemap/blob/master/index.js